File: pyjl/tests/tests_new/subscript_test.py

if __name__ == "__main__":
    l = [1,2,3]
    b = ["a", "b", "c"]
    x = 0
    ll = l[(x + 1):]
    x = 1
    lb = b[(x + 1):]

    assert ll == [2,3]
    assert lb == ["c"]

File: tests/cases-py2many/subscript_test.jl
if abspath(PROGRAM_FILE) == @__FILE__
    l = [1, 2, 3]
    b = ["a", "b", "c"]
    x = 0
    @assert(l[x+2:end] == [2, 3])
    x = 1
    @assert(b[x+2:end] == ["c"])
    output = [1, 2, 3, 4, 5, 6]
    start = 1
    stop = 3
    @assert(output[begin:stop-start] == [1, 2])
    @assert(output[length(output):end] == [6])
    @assert(output[end] == 6)
    println("OK")
end
